src/modules/ods/ods.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { GetOdsCustomerResponse } from './dto/get-ods-customer-response.dto';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { DATABASE } from '@ukef/constants';
+import { DataSource } from 'typeorm';
+import { PinoLogger } from 'nestjs-pino';
+import {
+  ODS_ENTITIES,
+  OdsEntity,
+  OdsStoredProcedureInput,
+  OdsStoredProcedureOuputBody,
+  OdsStoredProcedureOutput,
+  OdsStoredProcedureQueryParams,
+} from './dto/ods-payloads.dto';
+
+@Injectable()
+export class OdsService {
+  constructor(
+    @InjectDataSource(DATABASE.ODS)
+    private readonly odsDataSource: DataSource,
+    private readonly logger: PinoLogger,
+  ) {}
+
+  /**
+   * Finds a customer in ODS based on the URN provided
+   * @param {string} urn The customer URN to search for
+   *
+   * @returns {Promise<GetOdsCustomerResponse>} The customer response
+   * @throws {InternalServerErrorException} If there is an error trying to find a customer
+   * @throws {NotFoundException} If no matching customer is found
+   */
+  async findCustomer(urn: string): Promise<GetOdsCustomerResponse> {
+    try {
+      const spInput = this.createOdsStoredProcedureInput(ODS_ENTITIES.CUSTOMER, { customer_party_unique_reference_number: urn });
+
+      const storedProcedureResult = await this.callOdsStoredProcedure(spInput);
+
+      const storedProcedureJson: OdsStoredProcedureOuputBody = JSON.parse(storedProcedureResult[0]?.output_body);
+
+      if (storedProcedureJson === undefined || storedProcedureJson?.status != 'SUCCESS') {
```

Comment:
```suggestion
      if (storedProcedureJson === undefined || storedProcedureJson?.status !== 'SUCCESS') {
```
(sorry I missed this first time round 😅)

Reply:
Can it not just be `storedProcedureJson?.status !== 'SUCCESS'`, is there a reason for an explicity undefined comparison?

Reply:
It was there for code readability showing that it can be undefined rather than any required programmatical check, I'm happy to remove

Reply:
If we can please to keep the code clean that will be great. Thanks